### PR TITLE
Devfs attributes

### PR DESCRIPTION
**Pull request by franciscozdo**
* Introduced `devfs_getdata()` which returns data holded by devfs node. Before data was holded in `vnode::v_data` but now vnode holds pointer to devfs node. (It is needed to perform `vop_getattr`.)
* Added attributes to devfs nodes.
* For now we cannot do seek on character devices.

### CODE REVIEW

File: sys/kern/tmpfs.c [hunk 1]
```diff
@@ -982,7 +982,6 @@ static int tmpfs_root(mount_t *mp, vnode_t **vp) {
 }
 
 static int tmpfs_init(vfsconf_t *vfc) {
-  vnodeops_init(&tmpfs_vnodeops);
```
Comment (xThaid) [inline sys/kern/tmpfs.c:985]: Why is this removed?

File: sys/kern/vfs.c [hunk 2]
```diff
@@ -47,8 +47,6 @@ static vnodeops_t vfs_root_ops = {.v_lookup = vfs_root_vnode_lookup};
 static int vfs_register(vfsconf_t *vfc);
 
 void init_vfs(void) {
-  vnodeops_init(&vfs_root_ops);
```
Comment (xThaid) [inline sys/kern/vfs.c:50]: Same here.

File: sys/kern/tmpfs.c (hunk 1 above)
Comment (franciscozdo) [reply sys/kern/tmpfs.c]: I removed it by mistake. Thanks!

File: sys/kern/dev_null.c [hunk 3]
```diff
@@ -39,22 +39,17 @@ static int dev_zero_read(vnode_t *v, uio_t *uio, int ioflag) {
...
 
-static vnodeops_t dev_null_vnodeops = {.v_open = vnode_open_generic,
-                                       .v_read = dev_null_read,
+static vnodeops_t dev_null_vnodeops = {.v_read = dev_null_read,
                                        .v_write = dev_null_write};
 
-static vnodeops_t dev_zero_vnodeops = {.v_open = vnode_open_generic,
-                                       .v_read = dev_zero_read,
+static vnodeops_t dev_zero_vnodeops = {.v_read = dev_zero_read,
                                        .v_write = dev_zero_write};
```
Comment (pj1031999) [inline sys/kern/dev_null.c:46]: Could you format it?
```
static vnodeops_t dev_null_vnodeops = {
...
};
```

File: sys/kern/dev_vga.c [hunk 4]
```diff
@@ -56,8 +54,7 @@ static int videomode_read(vnode_t *v, uio_t *uio, int ioflag) {
   return uiomove_frombuf(buffer, RES_CTRL_BUFFER_SIZE, uio);
 }
 
-static vnodeops_t videomode_vnodeops = {.v_open = vnode_open_generic,
-                                        .v_read = videomode_read,
+static vnodeops_t videomode_vnodeops = {.v_read = videomode_read,
                                         .v_write = videomode_write};
```
Comment (pj1031999) [inline sys/kern/dev_vga.c:58]: Same.

File: sys/kern/devfs.c [hunk 5]
```diff
@@ -36,10 +42,16 @@ static devfs_mount_t devfs = {
   .lock = MTX_INITIALIZER(0), .next_ino = 2, .root = {}};
 static vnode_lookup_t devfs_vop_lookup;
 static vnode_readdir_t devfs_vop_readdir;
+static vnode_getattr_t devfs_vop_getattr;
 static vnodeops_t devfs_vnodeops = {.v_lookup = devfs_vop_lookup,
                                     .v_readdir = devfs_vop_readdir,
+                                    .v_getattr = devfs_vop_getattr,
                                     .v_open = vnode_open_generic};
```
Comment (pj1031999) [inline sys/kern/devfs.c:49]: Same.

File: sys/kern/devfs.c [hunk 6]
```diff
@@ -74,12 +86,41 @@ static int devfs_add_entry(devfs_node_t *parent, const char *name,
...
+void devfs_add_default_vops(vnodeops_t *vops) {
+  if (vops->v_open == NULL)
+    vops->v_open = vnode_open_generic;
+
+  if (vops->v_access == NULL)
+    vops->v_access = vnode_access_generic;
+
+  if (vops->v_getattr == NULL)
+    vops->v_getattr = devfs_vop_getattr;
+}
```
Comment (pj1031999) [inline sys/kern/devfs.c:122]: Could you make it static or even static inline? I don't see that function outside of devfs.c.

File: sys/kern/devfs.c [hunk 7]
```diff
@@ -89,7 +130,13 @@ int devfs_makedev(devfs_node_t *parent, const char *name, vnodeops_t *vops,
...
     return error;
 
-  dn->dn_vnode = vnode_new(V_DEV, vops, data);
+  devfs_add_default_vops(vops);
+  vnodeops_init(vops);
+
+  dn->dn_vnode = vnode_new(V_DEV, vops, dn);
+  dn->dn_data = data;
+  dn->dn_mode =
+    S_IFCHR | S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
```
Comment (pj1031999) [inline sys/kern/devfs.c:139]: `S_IWOTH` – others have write permission. Is it ok?

File: sys/kern/dev_null.c (hunk 3 above)
Comment (franciscozdo) [reply sys/kern/dev_null.c]: done

File: sys/kern/devfs.c (hunk 6 above)
Comment (franciscozdo) [reply sys/kern/devfs.c]: Done.

File: sys/kern/devfs.c (hunk 7 above)
Comment (franciscozdo) [reply sys/kern/devfs.c:141]: We have decided that it is ok for now.

File: sys/kern/dev_null.c (hunk 3 above)
Comment (franciscozdo) [reply sys/kern/dev_null.c]: Oops. It is done by `make format`...

Comment (franciscozdo) [reply sys/kern/dev_null.c:46]: I will leave it as it was

**Review by xThaid — APPROVED** [review]
Looks good now.

**Review by cahirwpz — APPROVED** [review]
LGTM
